en/docs/learn/using-opa-policies-for-adaptive-auth.md

```diff
@@ -0,0 +1,168 @@
+
+    ```
+    ./opa run --server
+    ```
+
+2. Write an OPA policy as shown in the example below. The policy below permits user authentication based on role, and allows only users that belong to the 'manager' or 'admin' role to log in. 
+
+    ```
+    package play.policy
+    import input
+    default permit = false
+     
+    Permit {
+        role = [“manager” , “admin”]
+        user := input.user.roles[_]
+        user == role[_]
+    }
+    ```
+
+
+3. If you have data, first upload the data file into OPA using the following curl command. (This tutorial does not use data, therefore this step is not required for this example.) 
+
+    ```
+    curl -X PUT http://localhost:8181/v1/data/myapi/acl --data-binary @myapi-acl.json
+    ```
+
+
+4. Upload the policy to the OPA engine using the following command. 
+
+    ```
+    curl -X PUT http://localhost:8181/v1/policies/myapi --data-binary @ispolicy.rego
+    ```
+
+---
+
+## Set up the sample app
+
+Set up the service provider and WSO2 IS sample application for adaptive authentication if you have not done so already. For instructions on how to do this, see [Configuring a Service Provider for Adaptive Authentication](../../learn/configuring-a-service-provider-for-adaptive-authentication/).
+
+You can skip this step if you are using your own application.
+
+---
+
+## Configure the adaptive authentication script
+
+1. Log in to the management console using admin/admin credentials. 
+
+2. Under the **Service Providers** section, click **Add**. 
+
+    !!! tip
+    If you are using the WSO2 IS sample application, you can skip steps 2 and 3 and directly edit the service provider you created when configuring the sample app.
+
+3. Enter a name for the service provider under **Service Provider Name** and click **Register**.
+
+4. Expand **Local and Outbound Configuration** and click **Advanced Authentication**.
+
+5. Click **Templates** on the right side of the **Script Based Conditional Authentication** field, and add the following script.
+
+    ```
+    var errorPage = '';
+      var errorPageParameters = {
+         'status': 'Unauthorized',
+         'statusMsg': 'You need to be an admin or a manager'
+      };
+      var onLoginRequest = function(context) {
+         executeStep(1, {
+             onSuccess: function (context) {
+                invokeOPA('http://localhost:8181/v1/data/play/policy', {"context" :context,"ip_address": "111.222.333.333"},{"sendClaims" : "false", "sendRoles": "true"}, {
+                      onSuccess : function(context, data) {
+                        var permit = data.result.permit;
+                        Log.info('permit '+ permit);
+                        Log.info("Successfully posted data.");
+                if (permit) {
+                      executeStep(2);
+                             }else{
```

Comment:
let's format this code (to have spaces around 'else'

Reply:
fixed